pkg/util/loadbalancer/loadbalancer.go

```diff
@@ -28,3 +31,24 @@ func RemoveFrontendIPConfiguration(lb *armnetwork.LoadBalancer, resourceID strin
 func isFrontendIPConfigReferenced(fipConfig *armnetwork.FrontendIPConfiguration) bool {
 	return fipConfig.Properties.LoadBalancingRules != nil || fipConfig.Properties.InboundNatPools != nil || fipConfig.Properties.InboundNatRules != nil || fipConfig.Properties.OutboundRules != nil
 }
+
+func RemoveHealthProbe(lb *armnetwork.LoadBalancer, resourceID string) error {
+	newProbes := make([]*armnetwork.Probe, 0)
+
+	if lb.Properties.Probes != nil {
```

Comment:
Sorry, i could have been clearer what I meant with the null check.
Checking `if lb.Properties.Probes != nil` here doesn't actually do anything. That is becuase the `for`-loop that follows handles the nil value gracefully and just doesn't go into the loop at all. so the `range` keyword already has that nil-check builtin.

What I meant to suggest is to add a nil-check to
```
if strings.EqualFold(*probe.ID, resourceID) {
```
so it looks like:

```
if probe.ID != nil && strings.EqualFold(*probe.ID, resourceID) {
...
```